### PR TITLE
sway: patch version

### PR DESCRIPTION
**Pull request by colemickens**
```
$(nix-build build.nix -A sway-unwrapped)/bin/sway --version
# ...
sway version 1.4-c0811fcf8736961725f7aa848e740f38f2bdf6ed (2020-03-04 03:09:43 +0100, branch 'master')
```

seems a bit wordy, not sure the version is necessary.

cc @cole-h

### CODE REVIEW

File: pkgs/sway/default.nix [hunk 1]
```diff
@@ -25,6 +25,9 @@ stdenv.mkDerivation rec {
 
   postPatch = ''
     # replace the version
+    sed -i "s/\([ \t]\)version: '\(.*\)',/\1version: '\2-${metadata.rev} (${metadata.revdate}, branch \\\'${metadata.branch}\\\')',/" \
```
Comment (cole-h) [inline pkgs/sway/default.nix]: No longer unnecessarily wordy (just need to bring `lib` or `substring` into scope) :)

```suggestion
    sed -i "s/\([ \t]\)version: '\(.*\)',/\1version: '\2-${lib.substring 0 8 metadata.rev} (${metadata.revdate}, branch \\\'${metadata.branch}\\\')',/" \
```

Comment (colemickens) [reply pkgs/sway/default.nix]: I'm still going to drop the date too, it's too easy to go from commit hash -> revdate for any user that wants to, either via git, github, or looking at metadata.nix.

**Comment by cole-h** [comment]
We could bring the date format in line with upstream by doing something like:

```nix
postPatch = ''
    date=date -d '${metadata.revdate}' +'%b %d %Y'
    sed -i "s/\([ \t]\)version: '\(.*\)',/\1version: '\2-${lib.substring 0 8 metadata.rev} ($date, branch \\\'${metadata.branch}\\\')',/"
'';
```

**Comment by colemickens** [comment]
`sway version 1.4-c0811fcf (Mar 04 2020, branch 'master')`

I don't ever actually build from git by hand, does that match?
